src/main/java/pl/put/poznan/buildingInfo/rest/BuildingInfoController.java

```diff
@@ -37,10 +40,18 @@
     @RequestMapping(value = "/new", method = RequestMethod.POST, produces = "application/json")
     public String createBuilding(@RequestBody String payload) {
         Building building = gson.fromJson(payload, Building.class);
+        // verify if location ids corresponds to assumed assumptions for key structure
         if (!Janitor.verifyBuildingStructure(building)) {
             Response res = new Response("Couldn't create a new building");
             return gson.toJson(res);
         }
+        // verify if locations are unique is unique among all buildings
+        List<Building> buildings = new ArrayList<>();
+        buildingRepository.findAll().forEach(buildings::add);
+        if (!Janitor.verifyUniqueBuildingId(building,buildings)) {
```

Comment:
I think we can try to do it more optimal.
We can just try to get this builing ID from DB and if result is null then this ID is uniqe, otherwise the building has been already created.

Reply:
good point! 👍